interview-questions.md

```diff
@@ -5,56 +5,129 @@ Answer the following questions. First, without external resources. Challenge you
 
   Your answer:
+  State is where the data can change/update.
+  State should not be mutated directly; use setState().
+  Props are the data passed from the parent component that stores the state to the child component. Props within the child component should not change or be passed back up to the parent component.
 
   Researched answer:
+  State is data maintained inside a component. It's local/owned by that component.
+  The component itself will update the state using setState().
 
+  Props is data passed from the parent component. Props are read-only in the child component that receives them. Callback functions can also be passed, which can be executed inside the child to initiate an update.
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
   Your answer:
+  Hoisting is when a variable or function is 'hoisted' to the top of the file. That could lead to the value being undefined, if not used carefully. When used correctly, functions can be invoked at the top of the file even if the function was declared at the bottom of the file.
 
   Researched answer:
+  All variable declarations using 'var' are hoisted/lifted to the top of their functional/local scope (declared inside a function) or the top of their global scope (declared outside a function) regardless of where the actual declaration has been made.
+
+  Function declarations are hoisted at the very top, before variable declarations.
 
+  Only variable declarations are hoisted (i.e., num = 8). Variable declarations must be written out before it's called, but must be initialized somewhere within the file.
 
 
 ## Looking Ahead: Terms for Next Week
 
 Research and define the following terms to the best of your ability.
 
 - React lifecycle methods
+Each component has a lifecyle that you can monitor and maniplulate during its three main phases. They are in order from start to finish.
+  1) Mounting - this is the phase where the component is created (your code and React internal's) and inserted into the DOM.
+  2) Updating - this is the phase where data is updated, whether it's state or props.
+  3) Unmounting - this is the final phase where the component is removed from the DOM.
+
+
 - API
+It stands for Application Programming Interface. It's a set of function that allows applications to access data and interact with software components, operating systems or microservices. It basically delivers a user's response to a system and sends the system's response back to the user.
+
 - event.preventDefault()
+This method tells the user agent not to use its default action if the event was not explicitly handled.
+
 - DOM events
+JS code can be exectued when an event occurs. You can add it in the HTML as an onclick or in React's render method.
+
+Events:
+
+When a user clicks the mouse
+When a web page has loaded
+When an image has been loaded
+When the mouse moves over an element
+When an input field is changed
+When an HTML form is submitted
+When a user strokes a key
```

Comment:
excellent! very well thought out answers, i can tell that you took your time to understand all of this.